### ISSUE TITLE
IDX10803: Unable to create to obtain configuration from: openid-configuration

### ISSUE THREAD

**Issue by chazt3n** [IdentityServer/IdentityServer3#2919]
- [x] I read and understood how to enable [logging](https://identityserver.github.io/Documentation/docsv2/configuration/logging.html)
### I'm trying to stand up an STS backed by AD and rather than being prompted for windows credentials, I immediately get this error page:

![image](https://cloud.githubusercontent.com/assets/3843179/15520481/4844bf10-21cc-11e6-87d2-e74afbd7cbd2.png)
## **IdServer Setup**

**Client**
![image](https://cloud.githubusercontent.com/assets/3843179/15520533/90eac7e6-21cc-11e6-9f62-9e8333caadfc.png)

**Startup**
![image](https://cloud.githubusercontent.com/assets/3843179/15520542/9b971a82-21cc-11e6-9003-4024566af4ed.png)
## **MVC Client Setup**

**Startup**
![image](https://cloud.githubusercontent.com/assets/3843179/15520672/1a38301a-21cd-11e6-99c9-2d4647dc8f1c.png)

**Controller**
![image](https://cloud.githubusercontent.com/assets/3843179/15520603/ccbebc32-21cc-11e6-9cc4-ddc78656814d.png)

I don't believe this is getting far enough to be logged
### Relevant parts of the log file

```
w3wp.exe Warning: 0 : 05/24/2016 14:13:03 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 ConsentStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 14:13:03 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 RefreshTokenStore not configured - falling back to InMemory
w3wp.exe Information: 0 : 05/24/2016 14:13:04 -05:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 05/24/2016 14:35:08 -05:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 05/24/2016 14:35:22 -05:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 05/24/2016 15:24:19 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Starting configuration.
w3wp.exe Information: 0 : 05/24/2016 15:24:19 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Adding WS-Federation endpoint
w3wp.exe Information: 0 : 05/24/2016 15:24:19 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Configuration done.
w3wp.exe Warning: 0 : 05/24/2016 15:24:19 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 AuthorizationCodeStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 15:24:19 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 TokenHandleStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 15:24:19 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 ConsentStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 15:24:19 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 RefreshTokenStore not configured - falling back to InMemory
w3wp.exe Information: 0 : 05/24/2016 15:25:34 -05:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 05/24/2016 15:26:00 -05:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 05/24/2016 15:37:00 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Starting configuration.
w3wp.exe Information: 0 : 05/24/2016 15:37:00 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Adding WS-Federation endpoint
w3wp.exe Information: 0 : 05/24/2016 15:37:00 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Configuration done.
w3wp.exe Warning: 0 : 05/24/2016 15:37:00 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 AuthorizationCodeStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 15:37:00 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 TokenHandleStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 15:37:00 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 ConsentStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 15:37:00 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 RefreshTokenStore not configured - falling back to InMemory
w3wp.exe Information: 0 : 05/24/2016 15:37:01 -05:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 05/24/2016 15:38:48 -05:00 [Information] (IdentityServer3.Core.Endpoints.DiscoveryEndpointController)
 Start discovery request
w3wp.exe Information: 0 : 05/24/2016 15:40:05 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Starting configuration.
w3wp.exe Information: 0 : 05/24/2016 15:40:05 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Adding WS-Federation endpoint
w3wp.exe Information: 0 : 05/24/2016 15:40:05 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Configuration done.
w3wp.exe Warning: 0 : 05/24/2016 15:40:05 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 AuthorizationCodeStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 15:40:05 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 TokenHandleStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 15:40:05 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 ConsentStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 15:40:05 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 RefreshTokenStore not configured - falling back to InMemory
w3wp.exe Information: 0 : 05/24/2016 16:03:44 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Starting configuration.
w3wp.exe Information: 0 : 05/24/2016 16:03:44 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Adding WS-Federation endpoint
w3wp.exe Information: 0 : 05/24/2016 16:03:45 -05:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Configuration done.
w3wp.exe Warning: 0 : 05/24/2016 16:03:45 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 AuthorizationCodeStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 16:03:45 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 TokenHandleStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 16:03:45 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 ConsentStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 05/24/2016 16:03:45 -05:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 RefreshTokenStore not configured - falling back to InMemory
w3wp.exe Information: 0 : 05/24/2016 16:03:47 -05:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 05/24/2016 16:03:52 -05:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 05/24/2016 16:21:04 -05:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering

```

**Comment by brockallen** [comment]
Sounds like you have client certs configured in IIS for the app. Again, unrelated to IdSvr :)

**Comment by chazt3n** [comment]
@brockallen Oh geez, I swear this isn't my first day at the office :disappointed: I have configured IIS to ignore the client certificate, now I'm fairly confident something in my setup is preventing my success

**Comment by chazt3n** [comment]
@brockallen OK, I enabled anonymous Auth and moved past this issue. Does that mean I've just forfeited security???

![image](https://cloud.githubusercontent.com/assets/3843179/15521403/2d20daa2-21d1-11e6-9e1d-609d52dae42c.png)

**Comment by brockallen** [comment]
The point of IdSvr is to do the authentication, not to have IIS do it.

**Comment by chazt3n** [comment]
@brockallen Thank you, I appreciate the hand holding

**Comment by chazt3n** [comment]
IdSrv was tabled at my company in May, and since then I've changed machines and had it put into our git repository.

pulling it down I had to create a new _localhost_ cert using the bat file here:
https://brockallen.com/2015/06/01/makecert-and-creating-ssl-or-signing-certificates/

My sites are showing "secured" in the browser, but I'm receiving these error messages again:

> IDX10803: Unable to create to obtain configuration from: 'https://localhost/IdentityServer/.well-known/openid-configuration'.
![image](https://cloud.githubusercontent.com/assets/3843179/21999676/ed69cc1c-dc00-11e6-8360-b790e1c8fcc6.png)


and 

>Response status code does not indicate success: 401 (Unauthorized).
![image](https://cloud.githubusercontent.com/assets/3843179/21999701/0897bff8-dc01-11e6-8af8-ecc895d4b3aa.png)

Logs spit this out
`w3wp.exe Information: 0 : 01/16/2017 14:58:41 -06:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Starting configuration.
w3wp.exe Information: 0 : 01/16/2017 14:58:41 -06:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Adding WS-Federation endpoint
w3wp.exe Information: 0 : 01/16/2017 14:58:41 -06:00 [Information] (Owin.WindowsAuthenticationAppBuilderExtensions)
 Configuration done.
w3wp.exe Warning: 0 : 01/16/2017 14:58:41 -06:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 AuthorizationCodeStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 01/16/2017 14:58:41 -06:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 TokenHandleStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 01/16/2017 14:58:41 -06:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 ConsentStore not configured - falling back to InMemory
w3wp.exe Warning: 0 : 01/16/2017 14:58:41 -06:00 [Warning] (IdentityServer3.Core.Configuration.IdentityServerServiceFactory)
 RefreshTokenStore not configured - falling back to InMemory
w3wp.exe Information: 0 : 01/16/2017 14:58:43 -06:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 01/16/2017 15:03:07 -06:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 01/16/2017 15:03:10 -06:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
w3wp.exe Information: 0 : 01/16/2017 15:27:01 -06:00 [Information] (IdentityServer3.Core.Endpoints.DiscoveryEndpointController)
 Start discovery request
w3wp.exe Information: 0 : 01/16/2017 15:27:41 -06:00 [Information] (IdentityServer3.Core.Endpoints.DiscoveryEndpointController)
 Start discovery request
w3wp.exe Information: 0 : 01/16/2017 15:29:37 -06:00 [Information] (IdentityServer3.Core.Endpoints.DiscoveryEndpointController)
 Start discovery request
w3wp.exe Information: 0 : 01/16/2017 15:29:41 -06:00 [Information] (IdentityServer3.Core.Endpoints.WelcomeController)
 Welcome page requested - rendering
`
I have this configuration only in the IdentityServer app
` <system.diagnostics>
    <trace autoflush="true" indentsize="4">
      <listeners>
        <remove name="Default" />
        <add name="myListener" type="System.Diagnostics.TextWriterTraceListener" initializeData="c:\logs\IdentityServer3.Samples.MinimalWebHost.log" />
      </listeners>
    </trace>
  </system.diagnostics>`

All three are hosted in IIS on my machine.

Have you got any ideas where I should start looking?
